resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/VolcanoTestsSuite.scala

```diff
@@ -151,6 +157,54 @@ private[spark] trait VolcanoTestsSuite { k8sSuite: KubernetesSuite =>
     )
   }
 
+  test("SPARK-38187: Run SparkPi Jobs with minCPU", k8sTestTag, volcanoTag) {
+    // Create a queue with 2 CPU, 3G memory capacity
+    createOrReplaceYAMLResource(VOLCANO_RES_LIMIT_QUEUE_YAML)
+    val jobNum = 4
+    (1 to jobNum).map { i =>
+      Future {
+        runJobAndVerify(
+          i.toString,
+          groupLoc = Option(GROUP_PREFIX),
+          queue = Option("queue-limit-resource"),
+          minCPU = Option("2"))
+      }
+    }
+    Eventually.eventually(TIMEOUT, INTERVAL) {
+      // Only 1 job running or all Succeeded
+      val pendingPods = getPods(role = "driver", GROUP_PREFIX, statusPhase = "Running")
+      assert(pendingPods.size <= 1)
+      // All jobs succeeded finally
+      val succeededPods = getPods(role = "driver", GROUP_PREFIX, statusPhase = "Succeeded")
+      assert(succeededPods.size === jobNum)
+    }
+    deleteYAMLResource(VOLCANO_RES_LIMIT_QUEUE_YAML)
```

Comment:
The deletion of the resources should be in `finally` or `after()` method.
Currently if a test fails the resources are not removed.